apps/ensindexer/ponder/src/ensdb-writer-worker.ts

```diff
@@ -0,0 +1,142 @@
+
+  // 1. Create ENSDb Client
+  const ensDbConnection = new EnsDbConnection();
+  const ensDbClient = ensDbConnection.connect({
+    schemaName: config.databaseSchemaName,
+    poolConfig: {
+      connectionString: config.databaseUrl,
+    },
+  });
+
+  logger.info("ENSDb Client connected");
+
+  // 2. Create ENSDb Query object for read operations
+  const ensDbQuery = new EnsDbQuery(ensDbClient);
+  // 3. Create ENSDb Mutation object for write operations
+  const ensDbMutation = new EnsDbMutation(ensDbClient);
+
+  /**
+   * Handle ENSIndexerPublicConfig Record
+   */
+  const handleEnsIndexerPublicConfigRecord = async () => {
+    // Read stored config and in-memory config.
+    // Note: we wrap read operations in pRetry to ensure all of them are
+    // completed successfully.
+    const [storedConfig, inMemoryConfig] = await pRetry(() =>
+      Promise.all([ensDbQuery.getEnsIndexerPublicConfig(), ensIndexerClient.config()]),
+    );
+
+    // Validate in-memory config object compatibility with the stored one,
+    // if the stored one is available
+    if (storedConfig) {
+      try {
+        validateENSIndexerPublicConfigCompatibility(storedConfig, inMemoryConfig);
+      } catch (error) {
+        const errorMessage =
+          "In-memory ENSIndexerPublicConfig object is not compatible with its counterpart stored in ENSDb.";
+
+        logger.error(error, errorMessage);
+
+        // Throw the error to terminate the ENSIndexer process due to
+        // found config incompatibility
+        throw new Error(errorMessage);
+      }
+    } else {
+      // Upsert ENSIndexerPublicConfig into ENSDb.
+      // Note: we wrap write operation in pRetry to ensure it can complete
+      // successfully, as there will be no other attempt.
+      await pRetry(() => ensDbMutation.upsertEnsIndexerPublicConfig(inMemoryConfig));
+      logger.info("ENSIndexer Public Config successfully stored in ENSDb.");
+    }
+  };
+
+  /**
+   * Handle Indexing Status Record Recursively
+   */
+  const handleIndexingStatusRecordRecursively = async () => {
+    try {
+      // Read in-memory Indexing Status.
+      const inMemoryIndexingStatus = await ensIndexerClient.indexingStatus();
+
+      // Check if Indexing Status is available.
+      if (inMemoryIndexingStatus.responseCode !== IndexingStatusResponseCodes.Ok) {
+        throw new Error("Indexing Status must be available.");
+      }
+
+      const { snapshot } = inMemoryIndexingStatus.realtimeProjection;
+      const { omnichainSnapshot } = snapshot;
+
+      // Check if Indexing Status is in expected status.
+      if (omnichainSnapshot.omnichainStatus === OmnichainIndexingStatusIds.Unstarted) {
+        throw new Error("Omnichain Status  must be different that 'Unstarted'.");
+      }
+
+      // Upsert ENSIndexerPublicConfig into ENSDb.
+      await ensDbMutation.upsertIndexingStatus(snapshot);
+
+      logger.info("Indexing Status successfully stored in ENSDb.");
```

Comment:
- add a once-per-process log at info level for "Indexing Status successfully stored in ENSDb."
- make this a trace-level call with something like "Indexing Status updated in ENSDb."

Reply:
In fact, there's no need to log the trace for the update event. The lack of error log implies the successful update.